dashboard/app/views/levels/editors/_gamelab.html.haml

```diff
@@ -80,6 +80,35 @@
   =f.label 'Disable code debugger (Console is still enabled)'
   = boolean_check_box f, :debugger_disabled
 
+.field
+  = f.label 'Immediately show the results of setup code changes in the playspace'
+  :ruby
+    selector = f.select :auto_run_setup, options_for_select([
+      ['none', nil],
+      ['Run the draw loop once', SharedConstants::GAMELAB_AUTORUN_OPTIONS.draw_loop],
+      ['Call drawSprites()', SharedConstants::GAMELAB_AUTORUN_OPTIONS.draw_sprites],
+      ['Run some other JavaScript', SharedConstants::GAMELAB_AUTORUN_OPTIONS.custom],
+    ], selected: @level.auto_run_setup)
+  = selector
+  = f.label 'Custom setup code'
+  = f.text_field :custom_setup_code
```

Comment:
This makes me a bit nervous.  While the Game Lab Sr. API set is pretty well fixed, we'll still be making changes to Jr. API.  There's a potential here to have lots of untested, unreviewed code written in levels that we depend on in shipped courses.

I don't have a better solution...

Reply:
I'd like to have some way to warn us about levels in "real" scripts that have custom helper code, custom blocks, or anything else like that. Would it be too heavy-handed to make such levels fail validation?